### ISSUE TITLE
Discussion of alternative salt windows repos available 

### ISSUE THREAD

**Issue by TheBigBear** [saltstack/salt#20245]
This is a much briefer paragraph to try to find out what the best salt repo for windows applications is?

Is the best way to use the current salt-winrepo (https://github.com/saltstack/salt-winrepo)? If so, please help discuss the right format and content of the windows sls files here: #20244 

But what about using the new and upcoming powershell 5 oneget (and chocolatey?) as one of salt's windows application repos? (see http://www.howtogeek.com/200334/windows-10-includes-a-linux-style-package-manager-named-oneget/ )

Or what about using the MS Powershell DSC (desired state configuration) way, the road chef would appear to be going down? (see http://www.slideshare.net/JulianDunn/chef-and-powershell-desired-state-configuration )

Or is it a mix of all of the above?

**Comment by ssgward** [comment]
Again, good questions. Thanks.

**Comment by lorengordon** [comment]
I vote for oneget! A built-in package manager for Windows! I wantz!

**Comment by TheBigBear** [comment]
OK, help me out, please. I am getting utterly confused. Just discovered that there is a salt module ("https://github.com/saltstack/salt/blob/develop/salt/modules/chocolatey.py") that i can bootstrap a chocolatey install unto my windwows minions and thereafter use the relevant 
```
salt 'my-winpc-1' chocolatey.install my-favourite-win-software-pkg
```
as well as list availble pkgs and remove or update them etc.
But how do I use the power of this salt chocolatey module to help me apply state to the windows minions with regards to what software pkgs they need to have installed, or updated, or removed etc?

**Comment by jfindlay** [comment]
@TheBigBear, You can always use the cmd.run state to run chocolatey commands.  The package system on windows is convoluted, (I mean, obviously since like the concept of package management was invented way back in ancient times) and currently about to change as microsoft has announced they'll support an official chocolatey-styled repo of their own.  You probably know more about this than I do.

It is likewise fraught in the way salt handles it.  The pkg state for windows could totally be updated to handle the multiple package management modules available for windows, and even add modules for package managers we don't have yet.  I think the best solution is support as many options as we can until the state of package management on windows converges into something more unified and standard than the way it is now.  That is the ideal solution.  Of course, implementation from the community is always welcome. :-)

**Comment by PatOShea** [comment]
TL;DR - OneGet and DSC are the future on windows. You'll use OneGet to download anything like apt-get, and DSC will be called to run the configuration steps. DSC enables reusing the same powershell configuration scripts across CM platforms (Chef/Puppet/Ansible/DSC itself). Using anything else will probably be reinventing the wheel.

@TheBigBear - I'm pretty late to this discussion, but I thought I could add a few thoughts. First, OneGet is the open source package manager "manager". Yes, that's not a typo. The windows version will be called PackageManagement on the next OS. The open source and shipped versions are the equivalent of Chromium and Chrome with Google's browser. As nice as chocolatey is now for EXEs, MSIs, gems, etc, OneGet will support more eventually and right now supports powershell modules from powershell galleries too.This is definitely the future and what windows has badly needed for a long time as @jfindlay alluded. The link below shows all the separate packaging platforms people want it to support. FearTheCowboy is an awesome resource about OneGet. He runs that project here on GitHub.https://github.com/OneGet/oneget/issues/77

As for DSC, that's configuration centric.  Some might disagree, but I feel the same way about that too. Other CM tools such as Chef, Puppet, and Ansible have already written modules/resources/etc to leverage DSC on the client. If you think about it, it's a sound idea.  Your tool interacts with "native" powershell modules which are used by multiple platforms to (Chef, Puppet, Ansible, DSC) and leverage that user base for improvements. Why roll your own Salt formula for configuring (IIS, SQL, Networking, etc) anything on windows when there is a pool of resources you can share the burden and benefit.

**Comment by dragon788** [comment]
Currently you have to use `chocolatey.installed` instead of `pkg.installed` but otherwise it should work similar to how you normally interact with Chocolatey. I'm in the process of rolling this out to a few of my machines to see if it ends up working consistently, but so far I've found one bug and am hunting down a couple others.

**Comment by lorengordon** [comment]
How about provider support for `pkg.installed`? That way, if you want to use chocolatey, just add `-provider: chocolatey` to the state...
